internal/service/synthetics/canary.go

```diff
@@ -717,6 +736,10 @@ func flattenCanaryRunConfig(canaryCodeOut *synthetics.CanaryRunConfigOutput) []i
 		"active_tracing":     aws.BoolValue(canaryCodeOut.ActiveTracing),
 	}
 
+	if v := envVars; v != nil {
```

Comment:
since we don't re-use the `v` variable instantiated, you can reduce this to `envVars` nil check e.g.
```suggestion
	if envVars != nil {
```